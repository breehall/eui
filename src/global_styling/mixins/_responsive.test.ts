/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { testCustomHook } from '../../test/internal';
import {
  useEuiBreakpoint,
  _isValidBreakpointArray,
  _validBreakpointSizes,
} from './_responsive';

const validBreakpointKeys = [
  undefined,
  '0',
  'xs',
  's',
  'm',
  'l',
  'xl',
  'infinity',
];

describe('_isValidBreakpointArray determines if the euiBreakpoint parameter array is invalid', () => {
  it('should return false if the array contains two elements and they are 0 and infinity', () => {
    expect(_isValidBreakpointArray(['0', 'infinity'])).toBeFalsy();
  });

  it('should return false if the array contains two elements and they both undefined', () => {
    expect(_isValidBreakpointArray([undefined, undefined])).toBeFalsy();
  });

  it('should return false if the array contains one element and it is 0', () => {
    expect(_isValidBreakpointArray(['0'])).toBeFalsy();
  });

  it('should return false if the array contains one element and it is undefined', () => {
    expect(_isValidBreakpointArray([undefined])).toBeFalsy();
  });

  it('should return false if the array contains one element and it is infinity', () => {
    expect(_isValidBreakpointArray(['infinity'])).toBeFalsy();
  });

  it('should return false if the array contains two elements and they are equal', () => {
    expect(_isValidBreakpointArray(['m', 'm'])).toBeFalsy();
  });

  it('should return false if the array contains two elements and the first element is larger than the second', () => {
    expect(_isValidBreakpointArray(['xl', 's'])).toBeFalsy();
  });
});

describe('test each possible two breakpoint combination', () => {
  const possibleTwoElementBreakpointCombinations = [];
  for (let i = 0; i < validBreakpointKeys.length; i++) {
    for (let inner = 1; inner < validBreakpointKeys.length; inner++) {
      // If the breakpoint is undefined, don't add quotes to make it a string
      possibleTwoElementBreakpointCombinations.push([
        validBreakpointKeys[i] === undefined
          ? undefined
          : `${validBreakpointKeys[i]}`,
        validBreakpointKeys[inner] === undefined
          ? undefined
          : `${validBreakpointKeys[inner]}`,
      ]);
    }
  }

  test.each(possibleTwoElementBreakpointCombinations)(
    'useEuiBreapoint returns a media query for two element breakpoint combinations (%s and %s)',
    (sizeA, sizeB) => {
      expect(
        testCustomHook(() =>
          useEuiBreakpoint([
            sizeA as _validBreakpointSizes,
            sizeB as _validBreakpointSizes,
          ])
        ).return
      ).toMatchSnapshot();
    }
  );
});

describe('test each possible single breakpoint combination', () => {
  validBreakpointKeys.forEach((size) => {
    it(`${size}`, () => {
      expect(
        testCustomHook(() => useEuiBreakpoint([size as _validBreakpointSizes]))
          .return
      ).toMatchSnapshot();
    });
  });
});
