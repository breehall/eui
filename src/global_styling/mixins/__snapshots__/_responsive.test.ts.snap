// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test each possible single breakpoint combination 0 1`] = `"@media only screen"`;

exports[`test each possible single breakpoint combination infinity 1`] = `"@media only screen"`;

exports[`test each possible single breakpoint combination l 1`] = `"@media only screen"`;

exports[`test each possible single breakpoint combination m 1`] = `"@media only screen"`;

exports[`test each possible single breakpoint combination s 1`] = `"@media only screen"`;

exports[`test each possible single breakpoint combination undefined 1`] = `"@media only screen"`;

exports[`test each possible single breakpoint combination xl 1`] = `"@media only screen and (min-width: 1200px)"`;

exports[`test each possible single breakpoint combination xs 1`] = `"@media only screen and (max-width: 574px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (0 and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (0 and infinity) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (0 and l) 1`] = `"@media only screen and (max-width: 992px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (0 and m) 1`] = `"@media only screen and (max-width: 768px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (0 and s) 1`] = `"@media only screen and (max-width: 575px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (0 and xl) 1`] = `"@media only screen and (max-width: 1200px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (0 and xs) 1`] = `"@media only screen and (max-width: 574px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (infinity and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (infinity and infinity) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (infinity and l) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (infinity and m) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (infinity and s) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (infinity and xl) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (infinity and xs) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (l and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (l and infinity) 1`] = `"@media only screen and (min-width: 992px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (l and l) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (l and m) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (l and s) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (l and xl) 1`] = `"@media only screen and (min-width: 992px) and (max-width: 9007199254740991px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (l and xs) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (m and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (m and infinity) 1`] = `"@media only screen and (min-width: 768px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (m and l) 1`] = `"@media only screen and (min-width: 768px) and (max-width: 1199px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (m and m) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (m and s) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (m and xl) 1`] = `"@media only screen and (min-width: 768px) and (max-width: 9007199254740991px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (m and xs) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (s and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (s and infinity) 1`] = `"@media only screen and (min-width: 575px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (s and l) 1`] = `"@media only screen and (min-width: 575px) and (max-width: 1199px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (s and m) 1`] = `"@media only screen and (min-width: 575px) and (max-width: 991px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (s and s) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (s and xl) 1`] = `"@media only screen and (min-width: 575px) and (max-width: 9007199254740991px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (s and xs) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (undefined and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (undefined and infinity) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (undefined and l) 1`] = `"@media only screen and (max-width: 992px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (undefined and m) 1`] = `"@media only screen and (max-width: 768px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (undefined and s) 1`] = `"@media only screen and (max-width: 575px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (undefined and xl) 1`] = `"@media only screen and (max-width: 1200px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (undefined and xs) 1`] = `"@media only screen and (max-width: 574px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xl and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xl and infinity) 1`] = `"@media only screen and (min-width: 1200px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xl and l) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xl and m) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xl and s) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xl and xl) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xl and xs) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xs and 0) 1`] = `"@media only screen"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xs and infinity) 1`] = `"@media only screen and (min-width: 0px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xs and l) 1`] = `"@media only screen and (min-width: 0px) and (max-width: 1199px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xs and m) 1`] = `"@media only screen and (min-width: 0px) and (max-width: 991px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xs and s) 1`] = `"@media only screen and (min-width: 0px) and (max-width: 767px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xs and xl) 1`] = `"@media only screen and (min-width: 0px) and (max-width: 9007199254740991px)"`;

exports[`test each possible two breakpoint combination useEuiBreapoint returns a media query for two element breakpoint combinations (xs and xs) 1`] = `"@media only screen"`;
